server/channels/testlib/helper.go

```diff
@@ -48,6 +48,11 @@ func NewMainHelper() *MainHelper {
 }
 
 func NewMainHelperWithOptions(options *HelperOptions) *MainHelper {
+	// Unset environment variables commonly set for development that interfere with tests.
+	os.Unsetenv("MM_SERVICESETTINGS_SITEURL")
+	os.Unsetenv("MM_SERVICESETTINGS_LISTENADDRESS")
+	os.Unsetenv("MM_SERVICESETTINGS_ENABLEDEVELOPER")
```

Comment:
This one caused me no end of trouble, as I often set `MM_*` into my environment to streamline testing. By simply unsetting, I guarantee that nothing downstream will reparse that back into the test, breaking the port on which the test helper listens, or changing the nature of assertions for errors that are enhanced when developer mode is enabled. (Also seen by Harrison at https://community.mattermost.com/private-core/pl/zfgerswe5fy89rirpokme91y6e)

We could probably just add a feature to config to ignore environment overrides altogether, but that proved to be more involved so I wanted to fix that too.